enhancements/api-review/custom-tags-aws.md

```diff
@@ -31,113 +36,325 @@ have the ability to apply user defined tags to many resources created by OpenShi
+
+The Infrastructure resource example to use spec for api changes
+
+```go
+// AWSPlatformSpec holds the desired state of the Amazon Web Services Infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type AWSPlatformSpec struct {
+    // Existing fields
+    ...
+    // ResourceTags is a list of additional tags to apply to AWS resources created for the cluster.
+    // See https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html for information on tagging AWS resources.
+    // AWS supports a maximum of 10 tags per resource. OpenShift reserves 5 tags for its use, leaving 5 tags
+    // available for the user.
+    // While ResourceTags field is mutable, items can not be removed.
+    // +kubebuilder:validation:MaxItems=10
+    // +optional
+    ResourceTags []AWSResourceTag `json:"resourceTags,omitempty"`
+}
+
+```
+
+```yaml
+spec:
+description: AWS contains settings specific to the Amazon Web Services Infrastructure provider.
+type: object
+properties:
+    resourceTags:
+    description: ResourceTags is a list of additional tags to apply to AWS resources created for the cluster. See https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html for information on tagging AWS resources.
+    type: array
+    maxItems: 10
+    items:
+        description: AWSResourceTag is a tag to apply to AWS resources created for the cluster.
+        type: object
+        required:
+            - key
+            - value
+        properties:
+            key:
+                description: key is the key of the tag
+                type: string
+                maxLength: 128
+                minLength: 1
+                pattern: ^[0-9A-Za-z_.:/=+-@]+$
+            value:
+                description: value is the value of the tag. Some AWS service do not support empty values. Since tags are added to resources in many services, the length of the tag value must meet the requirements of all services.
+                type: string
+                maxLength: 256
+                minLength: 1
+                pattern: ^[0-9A-Za-z_.:/=+-@]+$
+```
+
+### User Stories
 
-If `experimentalPropagateUserTags` is true, install validation will fail if there is any tag that starts with `kubernetes.io` or `openshift.io`.
+- As a security-conscious ROSA customer, I want to restrict the permissions granted to Red Hat in my AWS account by using
+AWS resource tags.Refer to https://issues.redhat.com/browse/SDE-1146.
 
-Add a new field `resourceTags` to `.status.aws` of the `infrastructure.config.openshift.io` type. Tags included in the
-`resourceTags` field will be applied to new resources created for the cluster. The `resourceTags` field will be populated by the installer only if the `experimentalPropagateUserTags` field is true.
+- As a cluster administrator of OpenShift, I would like to be able to add user-defined tags for AWS resources provisioned by an OCP cluster.
```

Comment:
To what end? Be specific about the story.

Reply:
Updated sentence to mention the maximum tags.